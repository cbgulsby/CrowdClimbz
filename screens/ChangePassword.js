import React, { useState } from 'react';
import { 
    StyleSheet, 
    View, 
    Text,
    SafeAreaView,
    TouchableOpacity,
    Button,
    TextInput,
    Alert,
} from 'react-native';

import firebase from '../firebase';

function passwordCheck(pass) {
    if (pass.length < 8) {
        return 1;
    }
    else return 0;
    //to be expanded upon later
}

async function changePassword(oldPass, newPass) {

    if (passwordCheck(newPass) == -1) {
        Alert.alert("Must enter valid password");
        return;
    }

    var auth = firebase.auth();
    var user = auth.currentUser;

    try {
        await user
            .updatePassword(newPass)
            .then(() => {
                Alert.alert("Password Updated!")
            })

        console.log("Account logged in");
    }
    catch (error) {
        console.log(error.toString());
    }
}


export default function ChangePassword({navigation}){   
    function send(val) {
        var dbh = firebase.firestore();
        if (!passwordCheck(val)){
            //dbh.collection('gyms').doc('testgym').update({name: val});
            //navigation.navigate('Profile');
            Alert.alert("Password Updated! (placeholder)");
        }    
    }

    const [value, onChangeText] = React.useState('Change Password');
    
    const [oldPass, setOldPass] = useState('');
    const [newPass, setNewPass] = useState('');

    return(
        <SafeAreaView style={styles.container}>
            <View>    
                
                <TextInput
                    secureTextEntry
                    style = { styles.inputContainer }
                    autoCapitalize = 'none'
                    placeholder = 'Current Password'
                    onChangeText = {
                        (oldPass) => setOldPass(oldPass)
                    }
                    value = { oldPass }
                />
                <TextInput
                    secureTextEntry
                    style = { styles.inputContainer }
                    autoCapitalize = 'none'
                    placeholder = 'New Password'
                    onChangeText = {
                        (newPass) => setNewPass(newPass)
                    }
                    value = { newPass }
                />

                <View style={{flexDirection: 'row', flex: 1}}>
                    <TouchableOpacity 
                        style={styles.button} 
                        onPress={() => navigation.navigate('Profile')}>
                        <Text> Cancel </Text>
                    </TouchableOpacity>

                    <TouchableOpacity 
                        style={styles.button} 
                        onPress={() => changePassword(oldPass, newPass)}>
                        <Text> Submit </Text>
                    </TouchableOpacity>      
                    
                </View>
            </View>

        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        paddingTop: 16,
        backgroundColor: '#118AB2',
        flex: 1
    },
    button: {
        backgroundColor: '#FFD166', 
        borderRadius: 4
    },
    inputContainer: {
        //flex: 11, 
        height: 40, 
        borderColor: '#073B4C', 
        borderWidth: 1, 
        paddingLeft: 5 
    },
    button: {
        backgroundColor: '#FFD166',
        borderColor: '#073B4C',
        borderWidth: 1,
        //borderRadius: 12,
        color: '#073B4C',
        fontSize: 24,
        fontWeight: 'bold',
        overflow: 'hidden',
        padding: 12,
        textAlign:'center',
        width: 200,
        justifyContent: 'center', 
        alignItems: 'center',
    }
});