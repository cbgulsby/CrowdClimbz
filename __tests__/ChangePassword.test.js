import React from 'react';
import renderer from 'react-test-renderer';

import ChangePassword from '../screens/ChangePassword';


test('renders correctly', () => {
const tree = renderer.create(<ChangePassword />).toJSON();
expect(tree).toMatchSnapshot();
});

// import passwordCheck from '../screens/ChangePassword';

// test('entered password too short', () => {
// 	expect(passwordCheck("hi")).toBe(1);
// });