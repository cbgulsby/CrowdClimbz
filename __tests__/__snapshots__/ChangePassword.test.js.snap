// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#4fb9ff",
      "flex": 1,
      "paddingTop": 16,
    }
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Change Password"
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Submit Changes
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
